phoenix-core/src/main/java/org/apache/phoenix/util/EncodedColumnsUtil.java

```diff
@@ -133,7 +133,7 @@ public static boolean useEncodedQualifierListOptimization(PTable table, Scan sca
          * Disabling this optimization for tables with more than one column family.
          * See PHOENIX-3890.
          */
-        return !scan.isRaw() && table.getColumnFamilies().size() <= 1 && table.getImmutableStorageScheme() != null
```

Comment:
This inadvertently caught local indexes too (which have 0 column families declared in their table definition)